uxarray/core/zonal.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+import dask.array as da
+
+
+def _compute_zonal_mean(uxda, data_mapping, latitudes, method="fast"):
+    # Select the appropriate function based on the data_mapping
+    if data_mapping == "n_face":
+        get_indices_func = uxda.uxgrid.get_faces_at_constant_latitude
+    else:
+        get_indices_func = uxda.uxgrid.get_edges_at_constant_latitude
+
+    # Create a result array based on whether Dask is being used
+    shape = uxda.shape[:-1] + (len(latitudes),)
+    if isinstance(uxda.data, da.Array):
+        # Create a Dask array for storing results
+        averages = da.empty(shape, dtype=uxda.dtype)
+    else:
+        # Create a NumPy array for storing results
+        averages = np.empty(shape, dtype=uxda.dtype)
+
+    for i, lat in enumerate(latitudes):
+        indices = get_indices_func(lat, method)
```

Comment:
@hongyuchen1030 @amberchen122 

With the API design I've decided to go with in #989, the accurate implementation in #785 will be specified by having `method='accurate'` instead of `method='fast'`

I can handle translating the implementation in #785 to meet the API design.

Reply:
Sounds good